ui/app/components/app/modals/qr-scanner/qr-scanner.component.js

```diff
@@ -104,15 +104,29 @@ export default class QrScanner extends Component {
   componentWillUnmount () {
     this.mounted = false
     clearTimeout(this.permissionChecker)
+    this.teardownCodeReader()
+  }
+
+  teardownCodeReader () {
     if (this.codeReader) {
       this.codeReader.reset()
+      this.codeReader.stop()
+      this.codeReader = null
     }
   }
 
   initCamera = async () => {
-    this.codeReader = new BrowserQRCodeReader()
+    // The `decodeFromInputVideoDevice` call prompts the browser to show
+    // the user the camera permission request.  We must then call it again
+    // once we receive permission so that the video displays.
+    // It's important to prevent this codeReader from being created twice;
+    // Firefox otherwise starts 2 video streams, one of which cannot be stopped
+    if (!this.codeReader) {
+      this.codeReader = new BrowserQRCodeReader()
+    }
     try {
       await this.codeReader.getVideoInputDevices()
+      this.checkPermissions()
```

Comment:
Kicking off this async all starts the component listening for the connection if permissions have not yet been given.